### PR TITLE
add control statements example

### PR DESCRIPTION
**Pull request by roiser**
A little example on control  statements, this may serve as the first example. It's rather easy and should give students confidence.

### CODE REVIEW

File: code/control/control.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#include <cstdio>
+
+int isodd(int i) { return i % 2; }
```
Comment (hageboeck) [inline code/control/control.cpp:3]:
```suggestion
bool isodd(int i) { return i % 2; }
```

File: code/control/solution/control.sol.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#include <cstdio>
+
+int isodd(int i) { return i % 2; }
```
Comment (hageboeck) [inline code/control/solution/control.sol.cpp:3]:
```suggestion
bool isodd(int i) { return i % 2; }
```

File: code/control/solution/control.sol.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    int num = numbers[i];
+    if (isodd(num)) {
+      sum_odd += num;
+    } else {
+      sum_eve += num;
+    }
+  }
+  printf("Sums: odd = %d, even = %d\n", sum_odd, sum_eve);
+
+  // range based loop
```
Comment (hageboeck) [inline code/control/solution/control.sol.cpp:22]:
```suggestion
  // range-based loop
```

File: code/control/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+## Instructions
+
+* Replace the C++98 style for loop with a range based loop
```
Comment (hageboeck) [inline code/control/README.md:3]:
```suggestion
* Replace the C++98-style for loop with a range-based loop
```

File: code/control/solution/control.sol.cpp [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  for (int i : numbers) {
+    isodd(i) ? sum_odd += i : sum_eve += i;
+  }
+  printf("Sums: odd = %d, even = %d\n", sum_odd, sum_eve);
+
+  // switch statement
+  sum_odd = 0;
+  sum_eve = 0;
+  for (int i : numbers) {
+    switch (isodd(i)) {
```
Comment (hageboeck) [inline code/control/solution/control.sol.cpp:46]:
```suggestion
    switch ( int(isodd(i)) ) {
```
Now I see why it was `int` ... 🙂

File: code/control/solution/control.sol.cpp [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  }
+  printf("Sums: odd = %d, even = %d\n", sum_odd, sum_eve);
+
+  // while loop
+  sum_odd = 0;
+  sum_eve = 0;
+  int i = 0;
+  while (i < nnums) {
+    int num = numbers[i++];
+    isodd(num) ? sum_odd += num : sum_eve += num;
```
Comment (hageboeck) [inline code/control/solution/control.sol.cpp:63]: Maybe we can try one-statement if bodies? It looks cleaner to me, but it's a matter of taste ...:
```suggestion
    if (isodd(num)) sum_odd += num;
    else sum_eve += num;
```

**Review by hageboeck — COMMENTED** [review]
Collection of random comments. Feel free to discard some/all.

File: code/control/control.cpp (hunk 1 above)
Comment (roiser) [reply code/control/control.cpp:3]: As you found out below the reason why it is int is that the move to the switch statement won't work. But, as we privately discussed, maybe a good solution is to have it bool here and in the description when asking the students to move to the switch statement tell them that the "isodd" function won't work anymore and they would need to slightly change it.

File: code/control/solution/control.sol.cpp (hunk 2 above)
Comment (roiser) [reply code/control/solution/control.sol.cpp:3]: I guess here we need to leave the int return type then.

File: code/control/solution/control.sol.cpp (hunk 5 above)
Comment (roiser) [reply code/control/solution/control.sol.cpp:46]: Actually I would leave the signature above then.

File: code/control/solution/control.sol.cpp (hunk 6 above)
Comment (roiser) [reply code/control/solution/control.sol.cpp:63]: Sebastien is using this syntax in his slides so wanted to repeat it here in the example code.

**Comment by sponce** [comment]
This is now superseded by #76 where I extended this branch with better cases for switch and while. @roiser if you can review it and confirm it works fine

**Pull request closed without merging**